packages/space-opera/src/components/mobile_view/mobile_view.ts

```diff
@@ -50,102 +47,107 @@ export class MobileView extends ConnectedLitElement {
   @internalProperty() arConfig: ArConfigState = {};
   @internalProperty() camera: Camera = INITIAL_CAMERA;
   @internalProperty() hotspots: HotspotConfig[] = [];
-  @internalProperty() gltf?: GltfModel;
+  @internalProperty() envImageUrl: string = '';
 
   @internalProperty() pipeId = window.location.search.replace('?id=', '');
-  @internalProperty() base = 'https://ppng.io/modelviewereditor';
-  @internalProperty() snippetPipeUrl = `${this.base}-state-${this.pipeId}`;
-  @internalProperty() updatesPipeUrl = `${this.base}-updates-${this.pipeId}`;
-  @internalProperty() mobilePingUrl = `${this.base}-ping-${this.pipeId}`;
-  @internalProperty() envPipeUrl = `${this.base}-env-${this.pipeId}`;
+  @internalProperty() mobilePingUrl = getPingUrl(this.pipeId);
 
-  stateChanged(state: State) {
-    this.config = getConfig(state);
-    this.arConfig = getArConfig(state);
-    this.hotspots = getHotspots(state);
-    this.camera = getCamera(state);
-  }
+  @internalProperty() toastClassName: string = '';
+  @internalProperty() toastBody: string = '';
 
-  async waitForState(envChanged: boolean) {
-    let partialState: any = {};
-    const response = await fetch(this.snippetPipeUrl);
-    if (response.ok) {
-      partialState = await response.json();
-    } else {
-      throw new Error('Something went wrong');
-    }
+  @internalProperty() sessionId = getRandomInt(1e+20);
+  @internalProperty() sessionUrl = getSessionUrl(this.pipeId, this.sessionId);
 
+  updateState(snippet: any, envChanged: boolean) {
     // The partialState env link correspondes to the editor's link.
     if (envChanged) {
-      partialState.config.environmentImage = undefined;
+      snippet.config.environmentImage = undefined;
     } else if (this.config.environmentImage) {
-      partialState.config.environmentImage = this.config.environmentImage;
+      snippet.config.environmentImage = this.config.environmentImage;
     }
 
-    reduxStore.dispatch(dispatchSetHotspots(partialState.hotspots));
-    reduxStore.dispatch(dispatchSetCamera(partialState.camera));
-    reduxStore.dispatch(dispatchSetConfig(partialState.config));
-    reduxStore.dispatch(dispatchArConfig(partialState.arConfig));
-
-    await this.updateComplete;
+    this.hotspots = snippet.hotspots;
+    this.arConfig = snippet.arConfig;
+    this.config = snippet.config;
+    this.camera = snippet.camera;
 
     // Send a new POST out for each scene-viewer button press
-    if (partialState.arConfig.ar) {
+    if (snippet.arConfig.ar) {
       const arButton =
           this.modelViewer?.shadowRoot!.getElementById('default-ar-button')!;
       // @ts-ignore
       arButton.addEventListener('click', (event: MouseEvent) => {
-        this.postSceneViewerBlob(this.currentBlob!);
+        post(this.currentBlob!, this.modelViewerUrl);
```

Comment:
I'd still like to remove all the @ts-ignore and if necessary replace them with more specific casts.